creatibutors: add spacing between icon and separator

jennur opened the pull request:
Closes https://github.com/inveniosoftware/invenio-app-rdm/issues/1736

- Added separator class to creatibutors-component in `SearchItemCreators` (to keep same markup in both places).
- Had to specify `.ui.items > .item .meta .creatibutors` for the styling to be preferred over default styling in the search-results case.
- Added spacing to separator class

## Screenshots
<img width="470" alt="Screenshot 2022-06-15 at 14 54 12" src="https://user-images.githubusercontent.com/21052053/173833154-494625a7-ba32-4af4-9d6f-e4f555aa4781.png">
<img width="473" alt="Screenshot 2022-06-15 at 14 54 00" src="https://user-images.githubusercontent.com/21052053/173833159-3f6f9b1f-966d-46ac-ba68-0b10ea0166c2.png">
<img width="720" alt="Screenshot 2022-06-15 at 14 53 49" src="https://user-images.githubusercontent.com/21052053/173833162-4112ac2f-7b02-45f7-9bc0-e29609ac4760.png">

kpsherva commented:
I am thinking if it is not an bit of an overkill to wrap the class around only the separators? a lot of typing for one semicolon... :)
alternative idea: what if we solve the matter once for all by implementing pseudoelement for horizontal list items, something like 
```less
.list
.item
&.separated:not(:last-child){
  &::after{
  content: ";"
  mergin-left: 0.1em;
  }
  }
  ```
  
  react usage
```javascript
  <List.Item className="separated" ... >
  ```
  
  html
  
  ````
  <div class="ui list" >
  <div class="item separated">
```

jennur commented:
## Updates
Removed the separator from the markup, and added it to the CSS within the `.separated`-class, to be used on list items.

![Screenshot 2022-06-20 at 10 59 47](https://user-images.githubusercontent.com/21052053/174566473-51b6a510-3866-40d9-9212-87168e622238.png)
![Screenshot 2022-06-20 at 10 59 35](https://user-images.githubusercontent.com/21052053/174566478-5d65e416-2cd1-40e8-9733-96d8c70e684c.png)
![Screenshot 2022-06-20 at 10 59 24](https://user-images.githubusercontent.com/21052053/174566481-22c3bba4-dab0-4f7e-a008-dc7a70d86328.png)
![Screenshot 2022-06-20 at 10 59 19](https://user-images.githubusercontent.com/21052053/174566483-d0a43677-2056-4fe9-b380-63819255b1df.png)

kpsherva commented on invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/globals/site.overrides:
```diff
@@ -571,3 +571,10 @@ dl.details-list {
   float: left;
 }
+
+.separated:not(:last-child)::after {
```
🥳

kpsherva commented on invenio_app_rdm/theme/assets/semantic-ui/less/invenio_app_rdm/theme/globals/site.overrides:
```diff
@@ -571,3 +571,10 @@ dl.details-list {
 }
+
+.separated:not(:last-child)::after {
+  content: ";";
```
could be as a variable in site.variables
`@listSeparator: ";"`
for easy overriding
